### PR TITLE
DAC6-975a

### PR DESCRIPTION
**Pull request by soljeggers**

### CODE REVIEW

File: app/controllers/DisclosureDetailsController.scala [hunk 1]
```diff
@@ -75,26 +73,28 @@ class DisclosureDetailsController @Inject() (
...
+                                                             id,
+                                                             disclosureDetails.firstInitialDisclosureMA.getOrElse(false)
+        ),
+        "reporterDetailsTaskListItem"      -> reporterDetailsItem(request.userAnswers, ReporterStatusPage, id),
+        "relevantTaxpayerTaskListItem"     -> relevantTaxpayersItem(request.userAnswers, RelevantTaxpayerStatusPage, id),
+        "associatedEnterpriseTaskListItem" -> associatedEnterpriseItem(request.userAnswers, AssociatedEnterpriseStatusPage, id),
+        "intermediariesTaskListItem"       -> intermediariesItem(request.userAnswers, IntermediariesStatusPage, id),
+        "othersAffectedTaskListItem"       -> othersAffectedItem(request.userAnswers, AffectedStatusPage, id),
+        "disclosureTaskListItem"           -> disclosureTypeItem(request.userAnswers, DisclosureStatusPage, id),
+        "userCanSubmit"                    -> userCanSubmit(request.userAnswers, id, disclosureDetails),
```
Comment (iservport) [inline app/controllers/DisclosureDetailsController.scala:91]: 👍

File: app/models/Submission.scala [hunk 2]
```diff
@@ -54,8 +55,13 @@ case class Submission(enrolmentID: String,
   val getDisclosureType: DisclosureType = disclosureDetails.disclosureType
 
   //Hide when MA = True, Reporter = intermediary, No relevant tax payers
-  val displayAssociatedEnterprises: Boolean =
-    !(disclosureDetails.initialDisclosureMA && reporterDetails.exists(_.isIntermediary) && taxpayers.isEmpty)
+  def displayAssociatedEnterprises(): Boolean =
+    disclosureDetails.disclosureType match {
+      case Dac6new =>
```
Comment (iservport) [inline app/models/Submission.scala:60]: 👍

File: test/helpers/TaskListHelperSpec.scala [hunk 3]
```diff
@@ -270,9 +270,59 @@ class TaskListHelperSpec extends ControllerMockFixtures with SpecBase with Scala
 
         val listOfPages = Seq(ReporterStatusPage, DisclosureStatusPage)
 
-        haveAllJourneysBeenCompleted(listOfPages, userAnswers, index, isInitialDisclosureMarketable = false) mustBe false
+        haveAllJourneysBeenCompleted(listOfPages, userAnswers, index, mockDisclosure.copy(firstInitialDisclosureMA = Some(false))) mustBe false
 
       }
+
+      "must return false if user is doing a REPLACEMENT of an NEW DISCLOSURE that IS MARKETABLE and " +
```
Comment (iservport) [inline test/helpers/TaskListHelperSpec.scala:277]: A previous test condition is: "must be true if user is doing REPLACEMENT DISCLOSURE OF ADDITIONAL DISCLOSURE & THAT IS NOT MARKETABLE but original DAC6NEW is marketable and has NOT started HALLMARKS or ARRANGEMENT DETAILS journey"  - The ticket says: For **ANY** replacement of an additional journey, InitialDisclosureMA flag should be defaulted to false. Is it a conflicting condition?

Comment (soljeggers) [reply test/helpers/TaskListHelperSpec.scala:277]: I believe so - there was a test condition that should have been removed way back when a change was made to how additional worked. I believe all the tests here to be correct now 😄

**Review by iservport — APPROVED** [review]
LGTM

**Pull request merged**